src/CodeGenerator/FactoryInterface.php

```diff
@@ -6,12 +6,16 @@
 
 use Psr\Container\ContainerInterface;
 
+/**
+ * @template T extends object
```

Comment:
Will this work in inheritance?

Reply:
Yepp should be no issue. The chance the this is implemented directly is almost zero.

Reply:
Oh yes, Implementing classes should use `@template-implements`. I adjusted the tests for the generated classes.

Reply:
Is this a working psalm annotation?
Shouldn't this be `@template T of object` instead?

Reply:
Correct, ith schould be `of`. This was TypeScript comming through.